### PR TITLE
Database logging

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR adds a config flag for turning on (or off) database logging.

### CODE REVIEW

File: config.yaml.defaults [hunk 1]
```diff
@@ -97,6 +97,9 @@ log:
     # log only those that trigger a warning / error
     api_calls: False
 
+    database_echo: False
```
Comment (stefanv) [inline config.yaml.defaults:100]: Please add comments explaining what these do.

"echo" is not very descriptive; perhaps use `log.database` and `log.database_pool`?

**Comment by stefanv** [comment]
Sorry, I don't mean to be annoying, but those comments cannot help a user understand what is being logged.

**Comment by stefanv** [comment]
From the page you linked:

```
sqlalchemy.engine - controls SQL echoing.
sqlalchemy.pool - controls connection pool logging.
```

So, while I haven't verified locally, looks like the first echoes SQL commands, and the second logs connections made to the database?

File: config.yaml.defaults [hunk 2]
```diff
@@ -97,10 +97,11 @@ log:
     # log only those that trigger a warning / error
     api_calls: False
 
-    # if true, log postgres database queries
+    # if true, enable SQL echoing.
     database: False
 
-    # set the debugging level (debug or info)
+    # if true, enable connection pool logging
+    # can also set the debugging level (debug or info)
```
Comment (stefanv) [inline config.yaml.defaults:104]: So can this flag be anything in `True, False, "debug", "info"`?

Comment (mcoughlin) [reply config.yaml.defaults]: I checked and True, False, and "debug" work. "info" did not for me so I took that off.

File: config.yaml.defaults [hunk 3]
```diff
@@ -97,6 +97,15 @@ log:
     # log only those that trigger a warning / error
     api_calls: False
 
+    # SQL logging
+    # see e.g. https://docs.sqlalchemy.org/en/20/core/engines.html#configuring-logging
+
+    # if True or "debug", enable SQL echoing.
+    database: False
+
+    # if true or "debug", enable connection pool logging
```
Comment (stefanv) [inline config.yaml.defaults:106]: `True`? And if `True` is just a synonym for `debug`, I'd remove that latter option. Preferably only one way to spell something! Then I'll merge.

Comment (mcoughlin) [reply config.yaml.defaults]: That does seem better.

**Comment by stefanv** [comment]
Thanks @mcoughlin!

**Pull request merged**